modules/component/src/push/push.pipe.ts

```diff
@@ -0,0 +1,132 @@
+import {
+  ChangeDetectorRef,
+  EmbeddedViewRef,
+  NgZone,
+  OnDestroy,
+  Pipe,
+  PipeTransform,
+  Type,
+} from '@angular/core';
+import {
+  NextObserver,
+  Observable,
+  PartialObserver,
+  Subject,
+  Unsubscribable,
+} from 'rxjs';
+import { distinctUntilChanged, map, withLatestFrom } from 'rxjs/operators';
+import {
+  CdAware,
+  CoalescingConfig as PushPipeConfig,
+  createCdAware,
+  setUpWork,
+} from '../core';
+
+/**
+ * @Pipe PushPipe
+ * @description
+ *
+ * The `ngrxPush` pipe serves as a drop-in replacement for the `async` pipe.
+ * It contains intelligent handling of change detection to enable us
+ * running in zone-full as well as zone-less mode without any changes to the code.
+ *
+ * The current way of binding an observable to the view looks like that:
+ *  ```html
+ *  {{observable$ | async}}
+ * <ng-container *ngIf="observable$ | async as o">{{o}}</ng-container>
+ * <component [value]="observable$ | async"></component>
+ * ```
+ *
+ * The problem is `async` pipe just marks the component and all its ancestors as dirty.
+ * It needs zone.js microtask queue to exhaust until `ApplicationRef.tick` is called to render all dirty marked components.
+ *
+ * Heavy dynamic and interactive UIs suffer from zones change detection a lot and can
+ * lean to bad performance or even unusable applications, but the `async` pipe does not work in zone-less mode.
+ *
+ * `ngrxPush` pipe solves that problem.
+ *
+ * Included Features:
+ *  - Take observables or promises, retrieve their values and render the value to the template
+ *  - Handling null and undefined values in a clean unified/structured way
+ *  - Triggers change-detection differently if `zone.js` is present or not (`detectChanges` or `markForCheck`)
```

Comment:
@BioPhoton PushPipe in zone-less mode uses detectChanges which triggers change detection synchronously. Isn’t it better to markForCheck and schedule CD (appref.tick) on at least the next requestAnimationFrame?

Reply:
Hi @eugene-stativka !

Definitely!

I summed everything related to this problem in a document here: 
https://hackmd.io/42oAMrzYReizA65AwQ7ZlQ

Please let me know the feedback here: https://github.com/ngrx/platform/issues/2441